app/assets/javascripts/admin/views/tasktype/task_type_create_view.js

```
@@ -1,175 +1,199 @@
+  teams: Array<APITeamType>,
+};
 
-    // Add 'required' attribute to select once it's supported
-    // https://github.com/davidstutz/bootstrap-multiselect/issues/620
-    if (_.isEmpty(formValues.settings.allowedModes)) {
-      Toast.error("Please provide at least one allowed mode.");
-      return;
-    }
+class TaskTypeCreateView extends React.PureComponent<Props, State> {
+  state = {
+    teams: [],
+  };
 
-    this.model.save(formValues).then(() => app.router.navigate("/taskTypes", { trigger: true }));
+  componentDidMount() {
+    this.fetchData();
+    this.applyDefaults();
   }
 
-  onRender() {
-    const teamSelectionView = new SelectionView({
-      collection: new TeamCollection(),
-      childViewOptions: {
-        modelValue() {
-          return `${this.model.get("name")}`;
-        },
-        defaultItem: { name: this.model.get("team") },
+  async applyDefaults() {
+    const defaultValues = {
+      settings: {
+        somaClickingAllowed: true,
+        branchPointsAllowed: true,
+        advancedOptionsAllowed: true,
       },
-      data: "amIAnAdmin=true",
-      name: "team",
-      required: true,
-    });
-    this.showChildView("team", teamSelectionView);
+    };
+    const taskType = this.props.taskTypeId ? await getTaskType(this.props.taskTypeId) : null;
 
-    this.ui.multiselect.multiselect();
+    const formValues = Object.assign({}, defaultValues, taskType);
+    this.props.form.setFieldsValue(formValues);
   }
 
-  onBeforeDestroy() {
-    this.ui.multiselect.multiselect("destroy");
+  async fetchData() {
+    this.setState({ teams: await getTeams() });
+  }
+
+  handleSubmit = e => {
+    e.preventDefault();
+    this.props.form.validateFields((err, formValues) => {
+      if (!err) {
+        if (this.props.taskTypeId) {
+          updateTaskType(this.props.taskTypeId, formValues);
+        } else {
+          createTaskType(formValues);
+        }
+
+        app.router.navigate("/taskTypes", { trigger: true });
```

Can we `await` `updateTaskType` and `createTaskType` before navigating back to taskTypes? Otherwise, errors will be silently swallowed.

---

Good catch :-)